pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/AbstractJsonPostOrPutServlet.java

```diff
@@ -0,0 +1,85 @@
+package com.redhat.pantheon.servlet;
+
+import com.redhat.pantheon.helper.TransformToPojo;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Abstract servlet class to provide common behaviour that accepts a json payload in the body of a POST or PUT
+ * web request, parses said payload into a java class, and then operates on the object. All the json
+ * deserialization logic is encapsulated in this class.
+ *
+ * Implementations should override the processPost of processPut methods (both are allowed) to implement
+ * specific logic for either case.
+ *
+ * @author Carlos Munoz
+ */
+public class AbstractJsonPostOrPutServlet<T> extends SlingAllMethodsServlet {
```

Comment:
I like this idea. However, I have a query - do you see any more servlets in the near term (1-2 sprints) where JSON payload would be expected? If no, my suggestion would be to table this until such a use case comes. Having an extra hierarchy for just one servlet seems non-optimal at the moment. What do you think?

Reply:
That's more a question for @benradey, @xdavidson and you really.
I'm happy to keep the PR on hold (somehow?) or close it if you feel we shouldn't add it... just scratching my coding itch :smile:

Reply:
Thanks :). I vote for holding it for now. @benradey, @xdavidson what do you think?

Reply:
I would vote to bring it in. We already have `AbstractJsonSingleQueryServlet` in the project which only has 1 child based on it currently, so there is some precedent for abstracting things away immediately. Also, if we sit on this PR, we might accidentally forget that it exists when we go to create our 2nd post api (whenever that may be).